src/rhsmlib/cloud/providers/azure.py

```diff
@@ -121,28 +132,75 @@ def is_likely_running_on_cloud(self):
 
         return probability
 
+    def get_api_versions(self) -> Union[List[str], None]:
+        """
+        This method tries to get list of API versions currently supported by Azure cloud provider
+        :return: list of API versions or None
+        """
+        api_versions_str = self._get_data_from_server("api_versions", self.AZURE_API_VERSIONS_URL)
+        api_versions = None
+        try:
+            api_versions_dict = json.loads(api_versions_str)
+        except TypeError as err:
+            log.error(f'Unable to decode Azure API versions: {err}')
+        else:
+            if 'apiVersions' in api_versions_dict:
+                api_versions = api_versions_dict['apiVersions']
+        return api_versions
+
+    def _fix_supported_api_version(self) -> Union[str, None]:
```

Comment:
can this be called more than once? should it be cached, e.g. using the `@functools.cache` decorator?

Reply:
This method would be called only in extreme situation. Use case for this situation is following: two old golden image is used for spinning up VM image on the Azure and Azure API used by old sub-man is not supported anymore. When this method is used, then `API_VERSION` is updated in the instance. It is de facto cached and there is no need to call this method again for instance of `AzureCloudProvider`. The `@functools.cache` can be used, when method has at least one argument that could be used as key in the cache. This method does not have any argument.